tests/cupyx_tests/test_cutensor.py

```diff
@@ -337,3 +337,133 @@ def test_contraction(self):
                              self.c, mode_c)
         cupy.testing.assert_allclose(self.c, self.c_ref,
                                      rtol=self.tol, atol=self.tol)
+
+
+@testing.parameterize(*testing.product({
+    'dtype_char': ['e', 'f', 'd', 'F', 'D'],
+    'shape': [(30, 40, 30, 35)],
+    'alpha': [0.5, 1.0],
+    'beta': [0.0, 1.0],
+}))
+@pytest.mark.skipif(not ct.available, reason='cuTensor is unavailable')
+class TestCuTensorIncontiguous:
+    _tol = {'e': 1e-3, 'f': 2e-6, 'd': 1e-12}
+
+    def make_random_array(self, shape, dtype):
+        r_dtype = dtype
+        if dtype == numpy.complex64:
+            r_dtype = numpy.float32
+        elif dtype == numpy.complex128:
+            r_dtype = numpy.float64
+        a = testing.shaped_random(shape, cupy, dtype=r_dtype, scale=1)
+        if dtype.char in 'FD':
+            a = a + 1j * \
+                testing.shaped_random(shape, cupy, dtype=r_dtype, scale=1)
+        return a
```

Comment:
This should not be necessary, `testing.shaped_random` can generate complex random arrays too, so just use it everywhere.